Sources/Db/APIs/MySQL.php

```diff
@@ -1984,18 +1984,24 @@ protected function __construct(array $options = [])
 			$this->prefixReservedTables();
 		}
 
-		$this->get_version();
-		$this->supports_pcre = version_compare($this->version, strpos($this->version, 'MariaDB') !== false ? '10.0.5' : '8.0.4', '>=');
-
 		// For backward compatibility.
 		if (!is_object(self::$db_connection)) {
 			self::$db_connection = $this->connection;
 		}
 
+		// At this point, if we don't have a connection, nothing else can be done.
+		if (empty($this->connection)) {
+			return;
+		}
+
```

Comment:
```suggestion
		// At this point, if we don't have a connection, nothing else can be done.
		if (empty($this->connection)) {
			return;
		}

		// For backward compatibility.
		if (!is_object(self::$db_connection)) {
			self::$db_connection = $this->connection;
		}

```

There's no point trying to set `self::$db_connection` if `$this->connection` is empty.